html/location.html

```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <title>Cacophonator Setup</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="/static/css/bootstrap.min.css">
+</head>
+
+<body>
+    {{template "navbar"}}
+
+    <div class="container">
+        <div 
+            id="resultMessage" class="alert alert-danger" 
+            {{if .ErrorEncountered}} 
+                style="display: block;" 
+            {{else}} 
+                style="display: none;" 
+            {{end}}> {{.ErrorMessage}}
+        </div>
+
+        <div class="container pt-5 pl-0">
+            <h2>Device Location<br></h2>
+        </div>
+        <hr>
+
+
+        <form id="locationForm" action="/location" method="POST">
+
+            <div class="row">
+              <div class="form-group col-md-4"> 
+                <label for="latitude">Latitude</label>
+                <input type="text" class="form-control" id="lat" name="latitude"
+                placeholder="Enter latitude" 
+                {{if ne .Location.Latitude 0.0}}
+                  value="{{.Location.Latitude}}"
+                {{end}}>                
+              </div>
+            </div>
+
+            <div class="row">
+              <div class="form-group col-md-4">
+                <label for="longitude">Longitude</label>
+                <input type="text" class="form-control" id="long" name="longitude"
+                placeholder="Enter longitude" 
+                {{if ne .Location.Longitude 0.0}}
+                  value="{{.Location.Longitude}}"
+                {{end}}>                
+              </div>
+            </div>
+
+          <button type="submit" class="btn btn-primary" onclick="handleLocationChange(event)" >Submit</button>
```

Comment:
If you make `/location` return HTML for the form on POST as well as GET then you don't need handleLocationChange any more. The browser can just post directly without the need for JavaScript.

Reply:
Ok, will give it a go.